libraries/rush-lib/src/logic/base/BaseInstallManager.ts

```diff
@@ -82,16 +92,26 @@ export abstract class BaseInstallManager {
     this.options = options;
 
     this._commonTempLinkFlag = LastLinkFlagFactory.getCommonTempFlag(rushConfiguration);
+    this.commonTempInstallFlag = LastInstallFlagFactory.getCommonTempFlag(rushConfiguration);
```

Comment:
The constructor creates `this.commonTempInstallFlag` here, however further down in this file we have `doInstallAsync()` doing this:

```ts
    // This marker file indicates that the last "rush install" completed successfully.
    // Always perform a clean install if filter flags were provided. Additionally, if
    // "--purge" was specified, or if the last install was interrupted, then we will
    // need to perform a clean install.  Otherwise, we can do an incremental install.
    const commonTempInstallFlag: LastInstallFlag = LastInstallFlagFactory.getCommonTempFlag(
      this.rushConfiguration,
      { npmrcHash: npmrcHash || '<NO NPMRC>' }
    );
```

And then **WorkspaceInstallManager.ts** does this:
```ts
        // Always save after pnpm rebuild to update timestamp of last install flag file.
        this.commonTempInstallFlag.create();
```

The flag file is a synchronization mechanism -- if `rush install` gets killed halfway through an operation, we need to determine where we left off, and whether node_modules is in a usable state or not. That is determined by the presence/absence of the flag file, and whether its contents match Rush's current configuration.  When dealing with synchronization, we can expect race conditions (code that "mostly" works but crashes randomly in situations that are very difficult to reproduce). Thus, the review of such code will generally focus on looking for guarantees of correctness, e.g. we delete the file before starting the operation, and our "finally" block guarantees its creation at the end, and the Rush commands that write this file use a reliable lock to prevent two processes from ever running simultaneously, etc.

This PR is changing the behavior of the flag file.  We need some code comments explaining the new behavior of the flag file: How do we track whether install scripts have been run yet or not?  Are we updating the flag file twice?  I wasn't able to figure that out easily from looking at the GitHub diff.